src/main/java/Main.java

```
@@ -0,0 +1,174 @@
+
+        ClientController controller = new ClientController(fixClient);
+
+        MessageListener<MessageGroupBatch> listener = (consumerTag, groupBatch) -> {
+            if (!controller.isRunning) controller.start(settings.autoStopAfter);
+
+            groupBatch.getGroupsList().forEach((group) -> {
+                try {
+                    if (group.getMessagesCount() != 1) {
+                        log.error("Message group contains more than 1 message", new IllegalArgumentException());
+                        throw new IllegalArgumentException();
+                    }
+                    AnyMessage message = group.getMessagesList().get(0);
+                    if (!message.hasRawMessage()) {
+                        log.error("Message in the group is not a raw message", new IllegalArgumentException());
+                        throw new IllegalArgumentException();
+                    }
+
+                    Session session = fixClient.getApplication().getSession();
+                    Message fixMessage = MessageUtils.parse(session, new String(message.getRawMessage().getBody().toByteArray(), StandardCharsets.UTF_8));
+                    session.send(fixMessage);
+
+                } catch (Exception e) {
+                    try {
+                        log.error("Failed to handle message group: " + MessageUtil.toPrettyString(group), e);
+                        MessageRouterUtils.storeEvent(eventRouter, rootEventId, "Failed to handle message group: " + MessageUtil.toPrettyString(group), "Error", e);
+                    } catch (InvalidProtocolBufferException invalidProtocolBufferException) {
+                        log.error("Failed to handle message group: ", invalidProtocolBufferException);
+                    }
+                }
+            });
+        };
+
+        try {
+            SubscriberMonitor monitor = messageRouter.subscribe(listener, INPUT_QUEUE_ATTRIBUTE);
+            if (monitor != null) {
```

We should throw an exception if it's `null`